lib/elixir/test/elixir/module/types/integration_test.exs

```diff
@@ -520,6 +523,18 @@ defmodule Module.Types.IntegrationTest do
       assert_warnings(files, warnings, consolidate_protocols: true)
     end
 
+    test "String.Chars protocol dispatch on improper lists" do
+      files = %{
+        "a.ex" => """
+        defmodule FooBar do
+          def example1, do: to_string([?a, ?b | "!"])
+        end
+        """
+      }
+
+      assert_no_warnings(files, consolidate_protocols: true)
+    end
```

Comment:
Ideally, if we can add this to one of the existing tests, it would be better, as they can be expensive!

Reply:
I couldn't find the perfect candidate, but maybe we can group regressions [here](https://github.com/elixir-lang/elixir/blob/main/lib/elixir/test/elixir/module/types/integration_test.exs#L1166) under a single module, replacing the test descriptions by comments?

Reply:
[d097960](https://github.com/elixir-lang/elixir/pull/14366/commits/d097960ccf256a25db5be16c67ea96ba101dec29)

Reply:
I like it!